barretenberg/cpp/src/barretenberg/vm2/simulation/interfaces/db.hpp

```diff
@@ -29,6 +29,7 @@ class ContractDBInterface {
     virtual void create_checkpoint() = 0;
     virtual void commit_checkpoint() = 0;
     virtual void revert_checkpoint() = 0;
+    virtual uint32_t get_checkpoint_id() const = 0;
```

Comment:
This is a bit gross (especially for the NAPI db which doesnt have a concept of checkpoint id) but the hint collecting db has no other way of 'seeing' the underlying db's checkpoint id. We use it inside checkpoint hints so it's currently required. I may try and just track the id as a private prop of `HintingContractsDB`(mirroring the way ts counts the id) if it doesn't turn out too ugly and remove this!

Reply:
I think I prefer to do it in private in `HintingContractsDB` (and not modify the interfaces)

Reasoning:
* this is only for keying the hints, in principle any id works as long as it works as a stack wrt checkpoints
* we'll have to use something different than checkpoint ids soon, because it actually only works "by chance" 😅  [it works because of how we use the DB but wouldnt work in the void]